encoding/encoding.go

```diff
@@ -100,54 +89,32 @@ func (e *Encoder) UnpackMessage(encodedBody []byte) ([]byte, MessageType, error)
 	}
 
 	flags := binary.BigEndian.Uint64(decoded[:8])
-	message, err := rsa.DecryptOAEP(
-		sha256.New(),
-		rand.Reader,
-		e.ownPrivKey,
-		decoded[8:],
-		rsaLabel,
-	)
+	r, err := age.Decrypt(bytes.NewReader(decoded[8:]), e.ownPrivKey)
 	if err != nil {
-		return nil, 0, fmt.Errorf("rsa.DecryptOAEP: %v", err)
+		return nil, 0, fmt.Errorf("failed to open decrypted stream: %v", err)
+	}
+	out := &bytes.Buffer{}
+	if _, err := io.Copy(out, r); err != nil {
+		return nil, 0, fmt.Errorf("failed to read from decrypted stream: %v", err)
 	}
 
-	return message, MessageType(flags), nil
+	return out.Bytes(), MessageType(flags), nil
 }
 
-func GenerateKey() (*rsa.PrivateKey, *rsa.PublicKey, error) {
-	private, err := rsa.GenerateKey(rand.Reader, rsaKeySize)
+func GenerateKey() (*age.X25519Identity, error) {
+	private, err := age.GenerateX25519Identity()
 	if err != nil {
-		return nil, nil, err
+		return nil, err
 	}
-	return private, &private.PublicKey, nil
+	return private, nil
```

Comment:
You can retrun result of "age.GenerateX25519Identity". There is no custom error text, so i don't see any reason for processing error here.